server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistSymbolsStep.java

```diff
@@ -0,0 +1,108 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.computation.step;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.sonar.api.resources.Qualifiers;
+import org.sonar.batch.protocol.output.BatchReport;
+import org.sonar.batch.protocol.output.BatchReportReader;
+import org.sonar.server.computation.ComputationContext;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.Maps.newHashMap;
+
+/**
+ * Nothing is persist for the moment. Only Symbols are read and not persist for the moment
+ */
+public class PersistSymbolsStep implements ComputationStep {
+
+  private static final String OFFSET_SEPARATOR = ",";
+  private static final String SYMBOLS_SEPARATOR = ";";
+
+  // Temporary variable in order to be able to test that symbols are well computed. Will only contains data from last processed file
+  private Map<Integer, StringBuilder> symbolsByLineForLastProcessedFile;
+
+  @Override
+  public String[] supportedProjectQualifiers() {
+    return new String[]{Qualifiers.PROJECT};
+  }
+
+  @Override
+  public void execute(ComputationContext context) {
+    int rootComponentRef = context.getReportMetadata().getRootComponentRef();
+    recursivelyProcessComponent(context, rootComponentRef);
+  }
+
+  private void recursivelyProcessComponent(ComputationContext context, int componentRef) {
+    BatchReportReader reportReader = context.getReportReader();
+    BatchReport.Component component = reportReader.readComponent(componentRef);
+    List<BatchReport.Symbol> symbols = reportReader.readComponentSymbols(componentRef);
+    processSymbols(component, symbols);
+
+    for (Integer childRef : component.getChildRefList()) {
+      recursivelyProcessComponent(context, childRef);
+    }
+  }
+
+  private void processSymbols(BatchReport.Component component, List<BatchReport.Symbol> symbols) {
+    symbolsByLineForLastProcessedFile = newHashMap();
+    if (!symbols.isEmpty()) {
+      int symbolId = 1;
+      for (BatchReport.Symbol symbol : symbols) {
+        processSymbolRange(symbol.getDeclaration(), symbolId);
+        for (BatchReport.Range reference : symbol.getReferenceList()) {
+          processSymbolRange(reference, symbolId);
+        }
+        symbolId++;
+      }
+    }
+  }
+
+  private void processSymbolRange(BatchReport.Range range, int symboleId){
+    int startLine = range.getStartLine();
+    if (startLine != range.getEndLine()) {
+      // TODO support symbols on multiple lines when source will be in compute, in order to be able to know the end line in this case
+      throw new IllegalStateException("To be implemented : Symbols on multiple lines are not supported for the moment");
```

Comment:
what is the behavior today ? shouldn't we have a symbol that goes until the end of the start line instead of throwing an exception ?